moztelemetry/dataset.py

```diff
@@ -41,6 +42,39 @@ def _group_by_size_greedy(obj_list, tot_groups):
     return groups
 
 
+def _group_by_equal_size(obj_list, tot_groups, threshold=pow(2, 32)):
+    sub_list = []
+    groups = []
+    priority = []
+    tmp_object = None
+    sub_list_size = 0
+
+    if tot_groups <= 1:
+        groups = _group_by_size_greedy(obj_list, tot_groups)
+        return groups
+
+    for obj in obj_list:
```

Comment:
List comprehensions are a nice compact way of doing this for-loop:

```python
priority = [(obj[key], obj) for obj in obj_list]
```

Reply:
The problem is obj_list is a list of dictionaries. It seems list comprehensions do not work in this case. In order to access the key of each dictionary, I thought I needed to index into the list first.

Reply:
It's doable if there's only one key being extracted. Here's a toy example.

```python
>>> test = [{'a': x, 'b': x**2} for x in range(3)]

>>> print(test)
[{'a': 0, 'b': 0}, {'a': 1, 'b': 1}, {'a': 2, 'b': 4}]

>>> [(x['b'], x) for x in test]
[(0, {'a': 0, 'b': 0}), (1, {'a': 1, 'b': 1}), (4, {'a': 2, 'b': 4})]
```

Reply:
This makes sense. It also works because you are able to specify a key that is known in the dictionary in the toy example. I will research more into list comprehensions with lists of dictionaries. However, it may be tricky to adapt the needed modifications to make a list comprehension work in this specific case.

Reply:
Are there other keys aside from the `size` key being used in the file summaries?

Reply:
It does work. Thanks for the clarification.